netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -258,6 +251,9 @@ public void operationComplete(ChannelFuture f) throws Exception {
       throw new IOException("Failed to bind", future.cause());
     }
     channel = future.channel();
+    Instrumented<SocketStats> listenSocket = new ListenSocket(channel);
+    listenSockets = ImmutableList.of(listenSocket);
```

Comment:
This is still racy, because `listenSockets` is still accessed on event loop threads, but assigned from an application thread.   You would need to make this volatile to not trigger the race detector.   

Can you show an example of what race you are trying to solve?